servicetalk-grpc-protoc/src/main/java/io/servicetalk/grpc/protoc/Generator.java

```diff
@@ -1639,6 +1645,57 @@ private TypeSpec newServiceInterfaceSpec(final State state, final boolean blocki
                 .addStatement("return $L", blocking ? BLOCKING_METHOD_DESCRIPTORS : ASYNC_METHOD_DESCRIPTORS)
                 .build());
 
+        // generate default service methods
+        if (defaultServiceMethods) {
+            final List<RpcInterface> interfaces = state.serviceRpcInterfaces.stream()
+                    .filter(intf -> intf.blocking == blocking)
+                    .collect(toList());
+            for (int i = 0; i < interfaces.size(); ++i) {
+                final RpcInterface rpcInterface = interfaces.get(i);
+                final MethodDescriptorProto methodProto = rpcInterface.methodProto;
+                final ClassName inClass = messageTypesMap.get(methodProto.getInputType());
+                final ClassName outClass = messageTypesMap.get(methodProto.getOutputType());
+                final String methodName = sanitizeIdentifier(methodProto.getName(), true);
+                final String methodPath = context.methodPath(state.serviceProto, methodProto).substring(1);
+                final int methodIndex = i;
+                final MethodSpec methodSpec = newRpcMethodSpec(inClass, outClass, methodName,
+                        methodProto.getClientStreaming(),
+                        methodProto.getServerStreaming(),
+                        !blocking ? EnumSet.of(INTERFACE) : EnumSet.of(INTERFACE, BLOCKING, SERVER_RESPONSE),
+                        printJavaDocs, (__, spec) -> {
+                            final String errorMessage = "\"Method " + methodPath + " is unimplemented\"";
+                            spec.addModifiers(DEFAULT).addParameter(GrpcServiceContext, ctx);
+                            spec.addAnnotation(Override.class);
+                            if (!blocking) {
+                                final ClassName returnType = methodProto.getServerStreaming() ? Publisher : Single;
+                                spec.addStatement("return $T.failed(new $T(new $T($T.UNIMPLEMENTED, $L)))",
+                                        returnType, GrpcStatusException, GrpcStatus, GrpcStatusCode, errorMessage);
+                            } else {
+                                if (!skipDeprecated && methodProto.getServerStreaming()) {
+                                    spec.addStatement("$T.super.$L(ctx, request, response)",
```

Comment:
We should always throw `UNIMPLEMENTED` exception instead of calling `super`.

Instead of
```java
        @Override
        default void testBiDiStream(GrpcServiceContext ctx, BlockingIterable<TestRequest> request,
                BlockingStreamingGrpcServerResponse<TestResponse> response) throws Exception {
            BlockingTestBiDiStreamRpc.super.testBiDiStream(ctx, request, response);
        }

        @Override
        default void testResponseStream(GrpcServiceContext ctx, TestRequest request,
                BlockingStreamingGrpcServerResponse<TestResponse> response) throws Exception {
            BlockingTestResponseStreamRpc.super.testResponseStream(ctx, request, response);
        }
```

We should generate:
```java
        @Override
        @Deprecated
        default void testBiDiStream(GrpcServiceContext ctx, BlockingIterable<TestRequest> request,
                                    GrpcPayloadWriter<TestResponse> responseWriter) throws Exception {
            throw new GrpcStatusException(new GrpcStatus(GrpcStatusCode.UNIMPLEMENTED, "Method grpc.netty.Tester/testBiDiStream is unimplemented"));
        }

        @Override
        @Deprecated
        default void testResponseStream(GrpcServiceContext ctx, TestRequest request,
                                        GrpcPayloadWriter<TestResponse> responseWriter) throws Exception {
            throw new GrpcStatusException(new GrpcStatus(GrpcStatusCode.UNIMPLEMENTED, "Method grpc.netty.Tester/testResponseStream is unimplemented"));
        }
```

Reply:
I would argue that this is not broken, it works as it should with the combination of flags. I definitely see your point about not redefining a default method when the base interface already has one. That is potentially easier said than done with javapoet.

Reply:
It will be ok to override both variants (deprecated and not) of `testBiDiStream` and `testResponseStream` and always throw `UNIMPLEMENTED` (no call to super). Then users will be able to override only non-deprecated variant and won't see any compilation problems asking to implement the deprecated one.